### PR TITLE
Add new Course Entitlement product class.

### PR DESCRIPTION
**Pull request by mjfrey**
This PR adds a new product class called "Course Entitlement".  It will allow learners to purchase a "course" rather than an individual course-run.

LEARNER-1728

### CODE REVIEW

File: ecommerce/core/constants.py [hunk 1]
```diff
@@ -21,6 +21,9 @@
 ENROLLMENT_CODE_SWITCH = 'create_enrollment_codes'
 ENROLLMENT_CODE_SEAT_TYPES = ['verified', 'professional', 'no-id-professional']
 
+# Course Entitlement contrant
+COURSE_ENTITLEMENT_CLASS_NAME = 'Course Entitlement'
```
Comment (clintonb) [inline ecommerce/core/constants.py]: `COURSE_ENTITLEMENT_PRODUCT_CLASS_NAME`

File: ecommerce/extensions/catalogue/tests/test_course_entitlement.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ProductClass = get_model('catalogue', 'ProductClass')
+
+
+class CourseEntitlementTest(TestCase):
+    """ Test course entitlement products."""
+
+    def test_course_entitlement_product(self):
+        """Test if a course entitlement product is properly created."""
+        course_entitlement_product_class, _ = ProductClass.objects.get_or_create(name=COURSE_ENTITLEMENT_CLASS_NAME)
+        course_entitlement_product = ProductFactory()
```
Comment (clintonb) [inline ecommerce/extensions/catalogue/tests/test_course_entitlement.py]: There isn't much value in validating anything besides the custom attributes. The model attributes are part of Django.

```python
product = ProductFactory(product_class=course_entitlement_product_class)
product.attr.course_key = 'foo-bar'
product.attr.certificate_type = 'verified'
product.attr.save()

product.refresh_from_db()
self.assertEqual(product.attr.course_key, 'foo-bar')
self.assertEqual(product.attr.certificate_type, 'verified')
```

**Review by clintonb — APPROVED** [review]

File: ecommerce/core/constants.py [hunk 3]
```diff
@@ -21,6 +21,9 @@
 ENROLLMENT_CODE_SWITCH = 'create_enrollment_codes'
 ENROLLMENT_CODE_SEAT_TYPES = ['verified', 'professional', 'no-id-professional']
 
+# Course Entitlement contrant
```
Comment (catong) [inline ecommerce/core/constants.py]: nit: typo "contrant"

**Review by staubina — APPROVED** [review]
:+1:

**Pull request merged**